core/trino-spi/src/main/java/io/trino/spi/ptf/TableArgument.java

```diff
@@ -102,6 +102,71 @@ public boolean isPassThroughColumns()
         return passThroughColumns;
     }
 
+    public static Builder tableArgumentBuilder()
+    {
+        return new Builder();
+    }
+
+    public static final class Builder
+    {
+        private Optional<QualifiedName> name;
+        private RowType rowType;
+        private List<String> partitionBy = List.of();
+        private List<SortItem> orderBy = List.of();
+        private boolean rowSemantics;
+        private boolean pruneWhenEmpty;
+        private boolean passThroughColumns;
+
+        private Builder() {}
+
+        public Builder name(Optional<QualifiedName> name)
+        {
+            this.name = name;
+            return this;
+        }
+
+        public Builder rowType(RowType rowType)
+        {
+            this.rowType = rowType;
+            return this;
+        }
+
+        public Builder partitionBy(List<String> partitionBy)
```

Comment:
I have a mixed feeling on not prefixing the setters with `set` or `with`. 
On one hand I like slick look of that. Yet I think it is not in-line with what we typically do in Trino.

Reply:
We do it here: https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/metadata/FunctionMetadata.java and here https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/metadata/Signature.java which is a significant amount of usage :) 
I'll leave them as-is unless you require otherwise.